ipc-addr.c

```diff
@@ -0,0 +1,21 @@
+#define _POSIX_C_SOURCE 200809L
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "ipc.h"
+
+int get_ipc_address(char *address, size_t size) {
+	const char *wayland_socket = getenv("WAYLAND_DISPLAY");
```

Comment:
Nit: can we rename this to `wayland_display`? `WAYLAND_SOCKET` is something else entirely, it's an already-opened FD picked up by libwayland.